devices/PAA5100JE.py

```diff
@@ -102,62 +127,98 @@ def _read(self, register: bytes, length: int =1):
         self.spi.write_readinto(send_buf, result)
+        time.sleep_us(1)
         self.select.off()
+        time.sleep_us(19)
         
         # Return the read result, skipping the first byte (which corresponds to the register)
         return result[1:] if length > 1 else result[1]
 
     def _bulk_write(self, data: bytearray):
         """Write a list of commands into registers"""
+        self.select.on()
         for x in range(0, len(data), 2):
             register, value = data[x : x + 2]
             self._write(register, value)
+            time.sleep_us(20)
+        self.select.off()
+        time.sleep_us(100)
                 
-    def read_registers(self, reg: bytes, buf: bytearray, len: int):
+    def read_registers(self, registers: bytes, buf: bytearray, len: int):
         """Read an array of data from the registers"""
         self.select.on()
-        self.spi.write(bytearray([reg]))
+        self.spi.write(bytearray([registers]))
+        time.sleep_us(100)
         # Read data from the register
         buf[:] = self.spi.read(len)
-        self.select.off()   
+        time.sleep_us(1)
+        self.select.off()  
+        time.sleep_us(19)
 
     def shut_down(self, deinitSPI:bool =True):
         """Shutdown the sensor"""
         self.select.off()
         time.sleep_ms(1)
         self.select.on()
-        time.sleep_ms(1)
-        self.reset.on()
         time.sleep_ms(60)
         self.write(self.firmware.REG_SHUTDOWN, 1)
         time.sleep_ms(1)
         self.select.off()
         time.sleep_ms(1)
         if deinitSPI:
             self.SPI.deinit()
+    
+    def read_register_buff(self, addrs: bytes, buff: bytes):
+        """
+        addrs < 128
+        """
+        self.select.on()
+        self.spi.write(addrs)
+        time.sleep_us(100)  # tSRAD
+        self.spi.readinto(buff)
+        time.sleep_us(1)  # tSCLK-NCS for read operation is 120ns
+        self.select.off()
+        time.sleep_us(19)  # tSRW/tSRR (=20us) minus tSCLK-NCS
+
+    def write_register_buff(self, addrs: bytes, data: bytes):
+        """
+        addrs < 128
+        """
+        # flip the MSB to 1:...
+        self.select.on()
+        self.spi.write(addrs)
+        self.spi.write(data)
+        time.sleep_us(20)  # tSCLK-NCS for write operation
+        self.select.off()
+        time.sleep_us(100)  # tSWW/tSWR (=120us) minus tSCLK-NCS. Could be shortened, but is looks like a safe lower bound
         
 class MotionDetector(Analog_input):
     """
     Using the Analog_input code to interface with 2 PAA5100JE sensors
     reading `x` (SPI2) and `y` (softSPI) separately.
     """
-    def __init__(self, reset: str, cs1: str, cs2: str,
-                 name='MotDet', threshold=1, calib_coef=1,
-                 sampling_rate=100, event='motion'):
+    def __init__(self, cs1: str, cs2: str,
+                 name='MotSen', event='motion', 
+                 calib_coef=1, threshold=1, sampling_rate=100):

```

Comment:
don't change the arguments so it is still compatible with the other sensor
ex: keep the `reset` argument but do not use it